### DISCUSSION TITLE
Add/create new script from a different script

### DISCUSSION THREAD

**Discussion by Dwyriel** [violentmonkey/violentmonkey#2122] [Q&A] [answered]
Hello everyone, I'm trying to figure out if there is a way to interact with Violentmonkey (and maybe other userscript injectors) to create a new script directly from code by either passing the script source code directly or pointing to an url containing it. 

My use case is simple: I'm trying to make a script that adds a button to a few specific pages (like the github or gitlabs file source page) that, when clicked, would take the contents of that file and send it to violentmonkey as a new script, very similar to what happens when you open a webpage that ends with `.user.js`.

I've managed to find a bad workaround by creating a new tab and opening the raw page url there adding a useless GET parameter ending with "script.user.js" that is ignored by the page, but picked up by Violentmonkey. This was the only solution I could find and it is very messy, is there a better way to do this?

**Comment by tophf** [comment, marked as answer]
Your solution is the only way, AFAIK.
